cmd/influxd/upgrade/upgrade.go

```diff
@@ -44,72 +79,293 @@ var options = struct {
 
 	// flags for target InfluxDB
 	target optionsV2
+
+	// verbose output
+	verbose bool
+
+	logPath string
 }{}
 
-func init() {
-	flags := Command.Flags()
+func NewCommand() *cobra.Command {
 
 	// source flags
 	v1dir, err := influxDirV1()
 	if err != nil {
 		panic("error fetching default InfluxDB 1.x dir: " + err.Error())
 	}
 
-	flags.StringVar(&options.source.metaDir, "v1-meta-dir", filepath.Join(v1dir, "meta"), "Path to 1.x meta.db directory")
-
 	// target flags
 	v2dir, err := fs.InfluxDir()
 	if err != nil {
 		panic("error fetching default InfluxDB 2.0 dir: " + err.Error())
 	}
 
-	flags.StringVar(&options.target.boltPath, "v2-bolt-path", filepath.Join(v2dir, "influxd.bolt"), "Path to 2.0 metadata")
+	// os-specific
+	var defaultSsPath string
+	if runtime.GOOS == "windows" {
+		defaultSsPath = filepath.Join(homeOrAnyDir(), "influxd-upgrade-security.cmd")
+	} else {
+		defaultSsPath = filepath.Join(homeOrAnyDir(), "influxd-upgrade-security.sh")
+	}
```

Comment:
We are skipping windows for now, so I would not include any additional complexity to support it